core/src/main/java/hudson/Util.java

```diff
@@ -890,6 +890,21 @@ public static String encode(@NonNull String s) {
         // queryMap[38] = queryMap[43] = queryMap[61] = true;
     }
 
+    private static final boolean[] fullUriMap = new boolean[123];
+    static {
+        String raw =
+                "               0123456789       ABCDEFGHIJKLMNOPQRSTUVWXYZ      abcdefghijklmnopqrstuvwxyz";
+        //  "!# %    $ ' &  *()+,-./          =:;< >?@                           _[\]^ `                          {|}~
+        //  ^--so these are encoded
```

Comment:
```suggestion
        String raw = "               0123456789       ABCDEFGHIJKLMNOPQRSTUVWXYZ      abcdefghijklmnopqrstuvwxyz";
        //            !"#$%&'()*+,-./0123456789:;<=>?@                          [\]^_`                          {|}~
        //  ^--so these are encoded
```

The idea with the comments in the static section for uriMap was to start from the ASCII table as a list and to display which ones will be kept and which ones will be encoded. With your proposal we lost this information as the characters are in an arbitrary order (and could be confusing).